packages/manager/core/request-tagger/__tests__/request-tagger.spec.ts

```diff
@@ -0,0 +1,319 @@
+    defineCurrentPage(defaultCurrentPage);
+
+    const secretVersion = 'secret-version';
+    const secretPage = 'secret-page';
+    const secretPattern = '.*/secret$';
+    addHeadersOverride(secretPattern, {
+      [Header.VERSION]: secretVersion,
+      [Header.PAGE]: secretPage,
+    });
+
+    const commonPage = 'common';
+    addHeadersOverride('.*/common$', {
+      [Header.PAGE]: commonPage,
+    });
+
+    const {
+      [Header.VERSION]: secretVersionHeader,
+      [Header.PAGE]: secretPageHeader,
+    } = getHeaders('/engine/apiv6/endpoint/secret');
+    expect(secretVersionHeader).toEqual(secretVersion);
+    expect(secretPageHeader).toEqual(secretPage);
+
+    const { [Header.PAGE]: commonPageHeader } = getHeaders(
+      '/engine/apiv6/endpoint/common',
+    );
+    expect(commonPageHeader).toEqual(commonPage);
+
+    const {
+      [Header.VERSION]: notSecretVersionHeader,
+      [Header.PAGE]: notSecretPageHeader,
+    } = getHeaders('/engine/apiv6/endpoint/not-secret');
+    expect(notSecretVersionHeader).toEqual(defaultApplicationVersion);
+    expect(notSecretPageHeader).toEqual(defaultCurrentPage);
+
+    removeHeadersOverride(secretPattern);
+
+    const {
+      [Header.VERSION]: noMoreSecretVersionHeader,
+      [Header.PAGE]: noMoreSecretPageHeader,
+    } = getHeaders('/engine/apiv6/endpoint/secret');
+    expect(noMoreSecretVersionHeader).toEqual(defaultApplicationVersion);
+    expect(noMoreSecretPageHeader).toEqual(defaultCurrentPage);
+
+    const { [Header.PAGE]: commonPageHeader2 } = getHeaders(
+      '/engine/apiv6/endpoint/common',
+    );
+    expect(commonPageHeader2).toEqual(commonPage);
+  });
+
+  it('allows to remove overriden headers without impacting other overrides', () => {
+    const secretVersion = 'secret-version';
+    const secretPage = 'secret-page';
+    const secretPattern = '.*/secret$';
+    addHeadersOverride(secretPattern, {
+      [Header.VERSION]: secretVersion,
+      [Header.PAGE]: secretPage,
+    });
+
+    const commonPage = 'common';
+    const commonPattern = '.*/common$';
+    addHeadersOverride(commonPattern, {
+      [Header.PAGE]: commonPage,
+    });
+
+    let headersOverrides = getHeadersOverride();
+    expect(Object.keys(headersOverrides)).toEqual(
+      expect.arrayContaining([secretPattern, commonPattern]),
+    );
+
+    expect(headersOverrides[secretPattern]).toMatchObject({
+      [Header.VERSION]: secretVersion,
+      [Header.PAGE]: secretPage,
+    });
+
+    expect(headersOverrides[commonPattern]).toMatchObject({
+      [Header.PAGE]: commonPage,
+    });
+
+    removeHeadersOverride(secretPattern);
+    headersOverrides = getHeadersOverride();
+
+    expect(Object.keys(headersOverrides)).toEqual(
+      expect.arrayContaining([commonPattern]),
+    );
+    expect(Object.keys(headersOverrides)).toEqual(
+      expect.not.arrayContaining([secretPattern]),
+    );
+
+    expect(headersOverrides[commonPattern]).toMatchObject({
+      [Header.PAGE]: commonPage,
```

Comment:
Too many tests in one test case, this should be split into different use cases, it's not a `unit` test